diff --git a/velox/type/Filter.h b/velox/type/Filter.h
index 82ecd8eb2..cc8e6b31b 100644
--- a/velox/type/Filter.h
+++ b/velox/type/Filter.h
@@ -1290,7 +1290,7 @@ class BigintValuesUsingBitmask final : public Filter {
   const int64_t max_;
 };
 
-class BigintValuesUsingBloomFilter final : public Filter {
+class BigintValuesUsingBloomFilter : public Filter {
  public:
   static int64_t numBlocks(int64_t capacity) {
     return SplitBlockBloomFilter::numBlocks(capacity, 0.01);
@@ -1301,7 +1301,7 @@ class BigintValuesUsingBloomFilter final : public Filter {
         blocks_(numBlocks(capacity)),
         filter_(blocks_) {}
 
-  bool testInt64(int64_t value) const final {
+  bool testInt64(int64_t value) const override {
     return filter_.mayContain(hash(value));
   }
 
@@ -1318,7 +1318,7 @@ class BigintValuesUsingBloomFilter final : public Filter {
   }
 
   bool testInt64Range(int64_t /*min*/, int64_t /*max*/, bool /*hasNull*/)
-      const final {
+      const override {
     return true;
   }
 
